src/humanPose/draw.js

```diff
@@ -1158,25 +1218,29 @@ function updateJointsAndBones(poseRenderInstance, poseObject, timestamp) {
 
         jointPositions[jointId] = jointPosition;
 
-        if (RENDER_CONFIDENCE_COLOR) {
-            let keys = getJointNodeInfo(poseObject, i);
-            // zero confidence if node's public data are not available
-            let confidence = 0.0;
-            if (keys) {
-                const node = poseObject.frames[keys.frameKey].nodes[keys.nodeKey];
-                if (node && node.publicData[JOINT_PUBLIC_DATA_KEYS.data].confidence !== undefined) {
-                    confidence = node.publicData[JOINT_PUBLIC_DATA_KEYS.data].confidence;
-                }
+        let keys = getJointNodeInfo(poseObject, i);
+        // zero confidence if node's public data are not available
+        let confidence = 0.0;
+        if (keys) {
+            const node = poseObject.frames[keys.frameKey].nodes[keys.nodeKey];
+            if (node && node.publicData[JOINT_PUBLIC_DATA_KEYS.data].confidence !== undefined) {
+                confidence = node.publicData[JOINT_PUBLIC_DATA_KEYS.data].confidence;
             }
+        }
+        jointConfidences[jointId] = confidence;
+
+        if (RENDER_CONFIDENCE_COLOR) {
+            
             poseRenderInstance.setJointConfidenceColor(jointId, confidence);
         }
     }
 
-    const pose = new Pose(jointPositions, timestamp, {poseObjectId: poseObject.uuid});
+    const poseHasParent = Boolean(poseObject.parent);
```

Comment:
It's less pretty but `!!poseObject.parent` is the more standard way of converting to a boolean since `Boolean` is a boxed type